dialoptions.go

```diff
@@ -35,6 +35,12 @@ import (
 	"google.golang.org/grpc/stats"
 )
 
+func init() {
+	internal.SetDefaultDialOption = func(opt ...DialOption) {
+		defaultDialOption = opt
```

Comment:
`defaultDialOption = append(defaultDialOption, opt...)` & rename to `AddDefaultDialOptions`?

Reply:
For tests we can add a `Set` (or a `Clear`, which seems better for cleanup).  There's a pretty big usability problem with `Set` for normal operation, since it only allows for a single user.